src/compile/mark/area.ts

```diff
@@ -11,80 +14,107 @@ export namespace area {
-    const xFieldDef = model.encoding().x;
+    applyColorAndOpacity(p, model);
+    applyMarkConfig(p, model, ['interpolate', 'tension']);
+    return p;
+  }
+  
+  function orient(orient: string): VgValueRef {
+    if (orient) {
+      return { value: orient };
+    }
+    return undefined;
+  }
+  
+  function x(fieldDef: FieldDef, scaleName: string, stack: StackProperties): VgValueRef {
     // x
     if (stack && X === stack.fieldChannel) { // Stacked Measure
-      p.x = {
-        scale: model.scaleName(X),
-        field: model.field(X, { suffix: '_start' })
+      return {
+        scale: scaleName,
+        field: field(fieldDef, { suffix: '_start' })
       };
-    } else if (isMeasure(xFieldDef)) { // Measure
-      p.x = { scale: model.scaleName(X), field: model.field(X) };
-    } else if (isDimension(xFieldDef)) {
-      p.x = {
-        scale: model.scaleName(X),
-        field: model.field(X, { binSuffix: '_mid' })
+    } else if (isMeasure(fieldDef)) { // Measure
+      return { scale: scaleName, field: field(fieldDef) };
+    } else if (isDimension(fieldDef)) {
+      return {
+        scale: scaleName,
+        field: field(fieldDef, { binSuffix: '_mid' })
       };
     }
+    return undefined;
+  }
 
+  function x2(fieldDef: FieldDef, scaleName: string, stack: StackProperties, orient: string): VgValueRef {  
     // x2
     if (orient === 'horizontal') {
       if (stack && X === stack.fieldChannel) {
-        p.x2 = {
-          scale: model.scaleName(X),
-          field: model.field(X, { suffix: '_end' })
+        return {
+          scale: scaleName,
+          field: field(fieldDef, { suffix: '_end' })
         };
       } else {
-        p.x2 = {
-          scale: model.scaleName(X),
+        return {
+          scale: scaleName,
           value: 0
         };
       }
     }
+    return undefined;
+  }
 
+  function y(fieldDef: FieldDef, scaleName: string, stack: StackProperties): VgValueRef {
     // y
-    const yFieldDef = model.encoding().y;
     if (stack && Y === stack.fieldChannel) { // Stacked Measure
-      p.y = {
-        scale: model.scaleName(Y),
-        field: model.field(Y, { suffix: '_start' })
+      return {
+        scale: scaleName,
+        field: field(fieldDef, { suffix: '_start' })
       };
-    } else if (isMeasure(yFieldDef)) {
-      p.y = {
-        scale: model.scaleName(Y),
-        field: model.field(Y)
+    } else if (isMeasure(fieldDef)) {
+      return {
+        scale: scaleName,
+        field: field(fieldDef)
       };
-    } else if (isDimension(yFieldDef)) {
-      p.y = {
-        scale: model.scaleName(Y),
-        field: model.field(Y, { binSuffix: '_mid' })
+    } else if (isDimension(fieldDef)) {
+      return {
+        scale: scaleName,
+        field: field(fieldDef, { binSuffix: '_mid' })
       };
     }
-
+    return undefined;
+  }
+  
+  function y2(fieldDef: FieldDef, scaleName: string, stack: StackProperties, orient: string): VgValueRef {  
```

Comment:
I think we need to pass in scale and if (!scale), then don't output `scaleName`.